src/librbd/librbd.cc

```diff
@@ -2571,7 +2571,7 @@ namespace librbd {
 
     bool discard_zero = ictx->config.get_val<bool>("rbd_discard_on_zeroed_write_same");
     if (discard_zero && mem_is_zero(bl.c_str(), bl.length())) {
```

Comment:
Not related, just wondering: why is `bl.is_zero()` not used here?

Reply:
Good catch -- looks like it's just left-over from the original write-same commit 3fe9ab5140f

Reply:
Fixed.